packages/node/test/integration/reject-install.spec.ts

```diff
@@ -86,53 +86,67 @@ describe("Node method follows spec - rejectInstall", () => {
       "sends acks back to A, A installs it, both nodes have the same app instance",
     () => {
       it("sends proposal with non-null initial state", async done => {
-        // A channel is first created between the two nodes
-        const multisigAddress = await getNewMultisig(nodeA, [
-          nodeA.publicIdentifier,
-          nodeB.publicIdentifier
-        ]);
-        expect(multisigAddress).toBeDefined();
-        expect(await getInstalledAppInstances(nodeA)).toEqual([]);
-        expect(await getInstalledAppInstances(nodeB)).toEqual([]);
-
-        let appInstanceId;
-
-        // second, an app instance must be proposed to be installed into that channel
-        const appInstanceInstallationProposalRequest = makeInstallProposalRequest(
-          nodeB.publicIdentifier
-        );
-
         nodeA.on(
-          NODE_EVENTS.REJECT_INSTALL,
-          async (msg: RejectProposalMessage) => {
-            expect((await getProposedAppInstances(nodeA)).length).toEqual(0);
-            done();
+          NODE_EVENTS.CREATE_CHANNEL,
+          async (data: NodeTypes.CreateChannelResult) => {
+            expect(await getInstalledAppInstances(nodeA)).toEqual([]);
+            expect(await getInstalledAppInstances(nodeB)).toEqual([]);
+
+            let appInstanceId;
+
+            // second, an app instance must be proposed to be installed into that channel
+            const appInstanceInstallationProposalRequest = makeInstallProposalRequest(
+              nodeB.publicIdentifier
+            );
+
+            nodeA.on(
+              NODE_EVENTS.REJECT_INSTALL,
+              async (msg: RejectProposalMessage) => {
+                expect((await getProposedAppInstances(nodeA)).length).toEqual(
+                  0
+                );
+                done();
+              }
+            );
+
+            // node B then decides to reject the proposal
+            nodeB.on(
+              NODE_EVENTS.PROPOSE_INSTALL,
+              async (msg: ProposeMessage) => {
+                confirmProposedAppInstanceOnNode(
+                  appInstanceInstallationProposalRequest.params,
+                  await getProposedAppInstanceInfo(nodeA, appInstanceId)
+                );
+
+                const rejectReq = makeRejectInstallRequest(
+                  msg.data.appInstanceId
+                );
+
+                // Node A should have a proposal in place before Node B rejects it
+                expect((await getProposedAppInstances(nodeA)).length).toEqual(
+                  1
+                );
+
+                await nodeB.call(rejectReq.type, rejectReq);
+
+                expect((await getProposedAppInstances(nodeB)).length).toEqual(
+                  0
+                );
```

Comment:
The code above here can be cleaner... e.g.,
```js
const proposalsA = await getProposedAppInstances(nodeA);
const proposalsB = await getProposedAppInstances(nodeB);
expect(proposalsA.length).toBe(0);
expect(proposalsB.length).toBe(0);
```

I'm just particularly concerned with the code cleanliness in these files because they are pretty critical tests and if they aren't approachable to outsiders / other team members, no one will bother reading them. Again, not urgent to fix in this PR particularly but I want to signal my perspective on this.

Reply:
Agreed!